FinSearchUnified/tests/Helper/StaticHelperTest.php

```diff
@@ -0,0 +1,175 @@
+                'ActivateFindologic' => true,
+                'ShopKey' => 'ABCD0815',
+                'ActivateFindologicForCategoryPages' => false,
+                'findologicDI' => false,
+                'isSearchPage' => false,
+                'isCategoryPage' => true,
+                'expected' => true
+            ],
+            'FINDOLOGIC is Active and CheckDirectIntegration is false; IsCategoryPage is false and IsSearchPage is true' => [
+                'ActivateFindologic' => true,
+                'ShopKey' => 'ABCD0815',
+                'ActivateFindologicForCategoryPages' => false,
+                'findologicDI' => false,
+                'isSearchPage' => true,
+                'isCategoryPage' => false,
+                'expected' => false
+            ],
+            'FINDOLOGIC is Active and CheckDirectIntegration is false; isCategoryPage is true and isSearchPage is false and ActivateFindologicForCategoryPages is true' => [
+                'ActivateFindologic' => true,
+                'ShopKey' => 'ABCD0815',
+                'ActivateFindologicForCategoryPages' => true,
+                'findologicDI' => false,
+                'isSearchPage' => false,
+                'isCategoryPage' => true,
+                'expected' => false
+            ]
+        ];
+    }
+
+    /**
+     * Method for testing useShopSearch method in StaticHelper class
+     *
+     * @dataProvider configDataProvider
+     * @param bool $isActive
+     * @param string $shopKey
+     * @param bool $isActiveForCategory
+     * @param bool $checkIntegration
+     * @param bool $isSearchPage
+     * @param bool $isCategoryPage
+     * @param bool $expectedResult
+     */
+    function testUseShopSearch(
+        $isActive,
+        $shopKey,
+        $isActiveForCategory,
+        $checkIntegration,
+        $isSearchPage,
+        $isCategoryPage,
+        $expectedResult
+    ) {
+        $configArray = [
+            'ActivateFindologic' => $isActive,
+            'ShopKey' => $shopKey,
+            'ActivateFindologicForCategoryPages' => $isActiveForCategory,
+            'IntegrationType' => $checkIntegration ? Constants::INTEGRATION_TYPE_DI : Constants::INTEGRATION_TYPE_API
+        ];
+
+        $sessionArray = [
+            'isSearchPage' => $isSearchPage,
+            'isCategoryPage' => $isCategoryPage,
+            'findologicDI' => $checkIntegration
+        ];
+
+        // Create Mock object for Shopware Session
+        Shopware()->Container()->set('session', $this->getMockBuilder('\Enlight_Components_Session_Namespace')
+            ->setMethods(null)
+            ->getMock());
+
+        // Create Mock object for Shopware Config
+        Shopware()->Container()->set('config', $this->getMockBuilder('\Shopware_Components_Config')
+            ->setMethods(null)
+            ->disableOriginalConstructor()
+            ->getMock());
+
+        if ($isSearchPage !== null) {
+            $this->saveSession($sessionArray);
```

Comment:
`saveSession` and `saveConfig` shouldn't be necessary anymore since we are using mocks now. Instead, mock the respective methods and explicitly set the return values.